src/components/SidePanel/SidePanel.jsx

```diff
@@ -0,0 +1,207 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'react-peek/prop-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import Overlay from '../Overlay/Overlay';
+import GripperVerticalIcon from '../Icon/GripperVerticalIcon/GripperVerticalIcon';
+import CrossIcon from '../Icon/CrossIcon/CrossIcon';
+import DragCaptureZone from '../DragCaptureZone/DragCaptureZone';
+import { getFirst, omitProps } from '../../util/component-types';
+
+const cx = lucidClassNames.bind('&-SidePanel');
+
+const { any, bool, func, oneOf, node, number, string } = PropTypes;
+
+class SidePanel extends React.Component {
+	constructor(...args) {
+		super(...args);
+		const [{ isExpanded, width }] = args;
+		this.state = {
+			isResizing: false,
+			width,
+			isExpanded,
+		};
+
+		this.handleResizeStart = this.handleResizeStart.bind(this);
+		this.handleResize = this.handleResize.bind(this);
+		this.handleResizeEnd = this.handleResizeEnd.bind(this);
+	}
+
+	handleResizeStart() {
+		this.startWidth = this.state.width;
+		this.setState({
+			isResizing: true,
+		});
+	}
+
+	handleResize({ dX }) {
+		this.setState({
+			width: this.startWidth + dX * (this.props.position === 'right' ? -1 : 1),
+		});
+	}
+
+	handleResizeEnd({ dX }, { event }) {
+		this.setState({
+			width: this.startWidth + dX * (this.props.position === 'right' ? -1 : 1),
+			isResizing: false,
+		});
+		this.props.onResize(this.startWidth - dX, { props: this.props, event });
+	}
+
+	componentDidUpdate(prevProps) {
+		if (prevProps.isExpanded !== this.props.isExpanded) {
+			setTimeout(() => {
```

Comment:
do we need to `clearTimeout` on this in `componentWillUnmount`?

Reply:
You're right, it's good form, just in case unmount happens between an event loop tick. In all likely hood it'll never happen, but i'll add it just in case ;)